zerver/tests/test_subs.py

```diff
@@ -1556,6 +1556,44 @@ def test_gather_subscriptions(self):
         self.assert_length(queries, 4, exact=True)
 
     @slow(0.15, "common_subscribe_to_streams is slow")
+    def test_never_subscribed_streams(self):
+        # type: () -> None
+        """
+        Check never_subscribed streams are fetched correctly and not include invite_only streams.
+        """
+        realm = get_realm("zulip.com")
+        streams = ["stream_%s" % i for i in range(10)]
+        for stream in streams:
+            create_stream_if_needed(realm, stream)
+        users_to_subscribe = ["othello@zulip.com", "cordelia@zulip.com"]
+        ret = self.common_subscribe_to_streams(
+            self.email,
+            streams,
+            dict(principals=ujson.dumps(users_to_subscribe)))
+        self.assert_json_success(ret)
+        ret = self.common_subscribe_to_streams(
+            self.email,
+            ["stream_invite_only_1"],
+            dict(principals=ujson.dumps(users_to_subscribe)),
+            invite_only=True)
+        self.assert_json_success(ret)
+        with queries_captured() as queries:
+            subscribed, unsubscribed, never_subscribed, email_dict = gather_subscriptions_helper(self.user_profile)
+        self.assertTrue(len(never_subscribed) >= 10)
+        invite_only_never_sub = False
+        for stream_dict in never_subscribed:
+            if stream_dict['name'] == "stream_invite_only_1":
+                invite_only_never_sub = True
```

Comment:
Rather than creating the invite_only_never_sub variable, why not just do 

self.assertFalse(stream_dict['name'] == "stream_invite_only_1")

in the loop?